lib/dictBuilder/zdict.rs

```diff
@@ -934,26 +923,19 @@ unsafe fn ZDICT_analyzeEntropy(
                     &mut litLengthCount,
                     &mut repOffset,
                     (srcBuffer as *const core::ffi::c_char).add(pos) as *const core::ffi::c_void,
-                    *fileSizes.offset(u as isize),
+                    *fileSize,
                     notificationLevel,
                 );
-                pos = pos.wrapping_add(*fileSizes.offset(u as isize));
-                u = u.wrapping_add(1);
+                pos = pos.wrapping_add(*fileSize);
             }
             if notificationLevel >= 4 {
                 if notificationLevel >= 4 {
                     eprintln!("Offset Code Frequencies :");
                 }
-                u = 0;
-                while u <= offcodeMax {
+                for u in 0..=offcodeMax as usize {
```

Comment:
just a note: we've seen some terrible codegen with inclusive ranges on loops in the past. I don't think think this is particularly performance-sensitive, but using `0..offcodeMax + 1` might optimize better.